next-docs/pages/world/balance.mdx

```diff
@@ -1 +1,226 @@
+import { CollapseCode } from "../../components/CollapseCode";
+
 # Balance
+
+ETH is stored in the `World` contract and owned by namespaces.
+Any ETH sent with a call to a `System` is kept on the `World`, and credited to the namescape to which the `System` belongs.
+This has two important effects:
+
+- The ETH balance is shared between the different `System` contracts in the same namespace of the same `World`.
+- The same `System` can be used in multiple `World` contracts, and in multiple namespaces in the same `World`, and the ETH balances remain separate.
+
+To know how much ETH a call has included a `System` can check [`_msgValue()`](https://github.com/latticexyz/mud/blob/main/packages/world/src/WorldContext.sol#L34-L36).
+
+To know how much ETH a specific namespace has you can look in `world:Balances` (the `Balances` table of the `world` namespace).
+
+<details>
+
+<summary>See this in action</summary>
+
+1. Have a MUD application running.
+   The easiest way to do this is to [run the template locally](/template/typescript/getting-started).
+
+1. Here we are not concerned with the client, so change to the `contracts` page.
+   Because we are not concerned with the client, all the file names will be relative to `.../packages/contracts`.
+
+   ```sh copy
+   cd packages/contracts
+   ```
+
+1. Create a `.env` file with:
+
+   - `PRIVATE_KEY` - the private key of an account that has ETH on the blockchain.
+   - `WORLD_ADDRESS` - the address of the `World` to which you add the namespace.
+
+   If you are using the template with a fresh `pnpm dev`, then you can use this `.env`:
+
+   ```sh filename=".env" copy
+   # Anvil default private key for the second account
+   # (NOT the account that deployed the World)
+   PRIVATE_KEY=0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d
+
+   # Address for the world we are extending
+   WORLD_ADDRESS=0x6e9474e9c83676b9a71133ff96db43e7aa0a4342
+   ```
+
+1. Create this file as `scripts/GetBalance.s.sol`.
+
+   ```solidity filename="GetBalance.s.sol" copy
+   // SPDX-License-Identifier: MIT
+   pragma solidity >=0.8.21;
+
+   import { Script } from "forge-std/Script.sol";
+   import { console } from "forge-std/console.sol";
+   import { Balances } from "@latticexyz/world/src/codegen/tables/Balances.sol";
+   import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+   import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+   import { WorldResourceIdLib } from "@latticexyz/world/src/WorldResourceId.sol";
+
+   contract GetBalance is Script {
+     function run() external {
+       address worldAddress = vm.envAddress("WORLD_ADDRESS");
+       StoreSwitch.setStoreAddress(worldAddress);
+       console.log("World at:", worldAddress);
+       ResourceId nsResource = WorldResourceIdLib.encodeNamespace(bytes14(""));
+       console.log("Namespace ID: %x", uint256(ResourceId.unwrap(nsResource)));
+       uint256 balance = Balances.get(nsResource);
```

Comment:
sorry to be pedantic on this, but let's try to avoid abbreviations wherever possible
```suggestion
       ResourceId namespaceResource = WorldResourceIdLib.encodeNamespace(bytes14(""));
       console.log("Namespace ID: %x", uint256(ResourceId.unwrap(namespaceResource)));
       uint256 balance = Balances.get(namespaceResource);
```